bots/nymphora/nymphora_main.lua

```diff
@@ -0,0 +1,613 @@
+	end
+
+	local bActionTaken = false
+
+	if nLastHarassUtil > object.nStunThreshold and skills.stun:CanActivate() then
+		if nTargetDistanceSq < 900 * 900 then
+			bActionTaken = core.OrderAbilityPosition(botBrain, skills.stun, vecPosInHalfSec)
+		end
+	end
+	if not bActionTaken then
+		if nLastHarassUtil > object.nHealThreshold and skills.heal:CanActivate() then
+			if nTargetDistanceSq < 750 * 750 then
+				bActionTaken = useHeal(botBrain, vecPosInHalfSec)
+			end
+		end
+	end
+	if not bActionTaken then
+		if bCanSee then
+			local itemSheepstick = core.GetItem("Item_Morph")
+			if itemSheepstick then
+				local nRange = itemSheepstick:GetRange()
+				if itemSheepstick:CanActivate() and nLastHarassUtil > object.nSheepstickThreshold then
+					if nTargetDistanceSq < (nRange * nRange) then
+						bActionTaken = core.OrderItemEntityClamp(botBrain, unitSelf, itemSheepstick, unitTarget)
+					end
+				end
+			end
+		end
+	end
+	if not bActionTaken then
+		local itemFrostfieldPlate = core.GetItem("Item_FrostfieldPlate")
+		if itemFrostfieldPlate ~= nil and itemFrostfieldPlate:CanActivate() then
+			if nLastHarassUtil > object.nFrostfieldThreshold then
+				local nRangeSQ = 700 * 700
+				if nRangeSQ > nTargetDistanceSq then
+					botBrain:OrderItem(itemFrostfieldPlate.object, "None")
+					bActionTaken = true
+				end
+			end
+		end
+	end
+	if not bActionTaken then
+		local itemPuzzleBox = core.GetItem("Item_Summon")
+		if itemPuzzleBox and itemPuzzleBox:CanActivate() then
+			if nLastHarassUtil > object.nPuzzleThreshold then
+				botBrain:OrderItem(itemPuzzleBox.object, "None")
+				bActionTaken = true
+			end
+		end
+	end
+
+	if not bActionTaken then
+		return object.harassExecuteOld(botBrain)
+	end
+end
+object.harassExecuteOld = behaviorLib.HarassHeroBehavior["Execute"]
+behaviorLib.HarassHeroBehavior["Execute"] = HarassHeroExecuteOverride
+
+------------
+-- Escape --
+------------
+function behaviorLib.CustomRetreatExecute(botBrain)
+	local unitSelf = core.unitSelf
+	local myPos = unitSelf:GetPosition()
+
+	local bActionTaken = false
+
+	if behaviorLib.lastRetreatUtil > object.nStunThreshold and skills.stun:CanActivate() then
+		if core.NumberElements(core.localUnits.EnemyHeroes) > 0 then
+			local unitTarget = nil
+			local closestDistance = 999999999
+			for _, unit in pairs(core.localUnits.EnemyHeroes) do
+				local Distance2DSq = Vector3.Distance2DSq(myPos, unit:GetPosition())
+				if Distance2DSq < closestDistance then
+					unitTarget = unit
+					closestDistance = Distance2DSq
+				end
+			end
+
+			bActionTaken = core.OrderAbilityPosition(botBrain, skills.stun, unitTarget:GetPosition())
+		end
+	end
+
+	return bActionTaken
+end
+
+----------
+-- Push --
+----------
+function behaviorLib.customPushExecute(botBrain)
+	local unitSelf = core.unitSelf
+
+	bActionTaken = false
+
+	if unitSelf:GetManaPercent() > 0.7 then
+		local creeps = core.localUnits["EnemyCreeps"]

```

Comment:
You should probably create a copy of this table, because currently if I'm not mistaken, you're adding allyCreeps into enemyCreeps, which could mess with other functions. That being said, it is refreshed every frame so..